modules/core/shared/src/main/scala/gem/HorizonsDesignation.scala

```diff
@@ -44,53 +44,41 @@ object HorizonsDesignation {
    * Designation for a comet, in the current apparition. Example: `C/1973 E1` for Kohoutek, yielding
    * the query string `NAME=C/1973 E1;CAP`.
    */
-  final case class Comet(des: String) extends HorizonsDesignation(s"NAME=$des;CAP")
-
-  object Comet {
-    val des: Comet @> String =
-      Lens((a, b) => a.copy(des = b), _.des)
-  }
+  @Lenses final case class Comet(des: String) extends HorizonsDesignation(s"NAME=$des;CAP")
+  @SuppressWarnings(Array("org.wartremover.warts.PublicInference"))
```

Comment:
`@Lenses` generates lenses for all case class fields. They show up in the companion object as values of the same name. Do `Comet.des` is the lens for the `des` field. Unfortuntely generated lenses don't have ascribed types, which causes WartRemover to complain. I opened an issue for this https://github.com/julien-truffaut/Monocle/issues/530 that should be easy to fix.

Reply:
Is it too cumbersome to create the `Lens` manually? Just curious of how much the macros do for us here, I'm not suggesting to do it by hand.

Reply:
Once we get that warning fixed it's still two lines of code per lens because our style guidlines require an explicit return type. You can see the available variations [here](http://julien-truffaut.github.io/Monocle/optics/lens.html).